cpp/src/utilities/cython.cpp

```diff
@@ -27,28 +28,159 @@ namespace cython {
 // Populates a graph_container_t with a pointer to a new graph object and sets
 // the meta-data accordingly.  The graph container owns the pointer and it is
 // assumed it will delete it on destruction.
-//
-// FIXME: Should local_* values be void* as well?
 void populate_graph_container(graph_container_t& graph_container,
-                              legacyGraphTypeEnum legacyType,
-                              raft::handle_t const& handle,
-                              void* offsets,
-                              void* indices,
+                              raft::handle_t& handle,
+                              void* src_vertices,
+                              void* dst_vertices,
                               void* weights,
-                              numberTypeEnum offsetType,
-                              numberTypeEnum indexType,
+                              void* vertex_partition_offsets,
+                              numberTypeEnum vertexType,
+                              numberTypeEnum edgeType,
                               numberTypeEnum weightType,
-                              int num_vertices,
-                              int num_edges,
-                              int* local_vertices,
-                              int* local_edges,
-                              int* local_offsets,
+                              int num_partition_edges,
+                              int num_global_vertices,
+                              int num_global_edges,
+                              int partition_row_size,  // pcols
+                              int partition_col_size,  // prows
                               bool transposed,
                               bool multi_gpu)
 {
   CUGRAPH_EXPECTS(graph_container.graph_ptr_type == graphTypeEnum::null,
                   "populate_graph_container() can only be called on an empty container.");
 
+  bool do_expensive_check{false};
+  bool hypergraph_partitioned{false};
+
+  raft::comms::comms_t const& communicator = handle.get_comms();
+  int const rank = communicator.get_rank();
+  int partition_row_rank = rank / partition_row_size;
+  int partition_col_rank  = rank % partition_row_size;
+
+  // Setup the subcommunicators needed for this partition on the handle
+  partition_2d::subcomm_factory_t<partition_2d::key_naming_t, int>
+    subcomm_factory(handle, partition_row_size);
+
+  // Copy the contents of the vertex_partition_offsets (host array) to a vector
+  // as needed by the partition_t ctor.
+  int* vertex_partition_offsets_array = reinterpret_cast<int*>(vertex_partition_offsets);
+  std::vector<int> vertex_partition_offsets_vect;  // vertex_t
+
+  for (int32_t i=0; i<((partition_row_size * partition_col_size)+1); ++i) {
+     vertex_partition_offsets_vect.push_back(vertex_partition_offsets_array[i]);
+     std::cout<<"VPO"<<i<<": "<<vertex_partition_offsets_vect[i]<<std::endl;
+  }
```

Comment:
```
std::vector<int> vertex_partition_offsets_vect(
  vertex_partition_offsets_array,
  vertex_partition_offsets_array + (col_comm_size * row_comm_size)+1);
```

Reply:
done in next commit. I thought there was a better way, thanks.